genie-common-internal/src/test/java/com/netflix/genie/common/internal/configs/CommonInternalAutoConfigurationIntegrationTest.java

```diff
@@ -0,0 +1,60 @@
+/*
+ *
+ *  Copyright 2018 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+package com.netflix.genie.common.internal.configs;
+
+import com.netflix.genie.common.internal.dto.v4.converters.JobDirectoryManifestProtoConverter;
+import com.netflix.genie.common.internal.dto.v4.converters.JobServiceProtoConverter;
+import com.netflix.genie.test.categories.IntegrationTest;
+import org.assertj.core.api.Assertions;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.springframework.boot.autoconfigure.AutoConfigurations;
+import org.springframework.boot.test.context.runner.ApplicationContextRunner;
+
+/**
+ * Tests for behavior of {@link CommonInternalAutoConfiguration}.
+ *
+ * @author mprimi
+ * @since 4.0.0
+ */
+@Category(IntegrationTest.class)
+public class CommonInternalAutoConfigurationIntegrationTest {
+    private final ApplicationContextRunner contextRunner = new ApplicationContextRunner()
+        .withConfiguration(
+            AutoConfigurations.of(
+                CommonInternalAutoConfiguration.class
+            )
+        )
+        .withPropertyValues(
+            "spring.jmx.enabled=false",
+            "spring.main.webApplicationType=none"
+        );
+
+    /**
+     * Test expected context.
+     */
+    @Test
+    public void testExpectedContext() {
+        this.contextRunner.run(
+            (context) -> {
+                Assertions.assertThat(context).hasSingleBean(JobServiceProtoConverter.class);
+                Assertions.assertThat(context).hasSingleBean(JobDirectoryManifestProtoConverter.class);
```

Comment:
Not sure what you mean..

Reply:
well you've proven with these tests that in the presence of no other configurations your auto configuration will generate the two desired beans. You've annotated them with `@ConditionalOnMissingBean` so you could do more tests which add user level configs (e.g. our internal wrapper `@Configuration` classes are user level configs) which provide the beans and make sure your default beans back off (aren't created). It's possibly overkill cause this is simple I just wanted to point out it's one of the main features of this testing.

I looked around and it looks like I hadn't done this for the Genie tests yet but I have done it for the internal DAS stuff if you want example or refer to the [Docs](https://docs.spring.io/spring-boot/docs/current/reference/html/boot-features-developing-auto-configuration.html#boot-features-test-autoconfig)

Reply:
Wouldn't that be testing Spring's code, rather than ours?
e.g. we don't test that the @NotNull annotation actually stop a null argument, or that @Slf4j actually prints our messages, etc. How is this any different?